packages/nimble-components/src/select/testing/select.pageobject.ts

```diff
@@ -0,0 +1,144 @@
+import {
+    keyEnter,
+    keyEscape,
+    keyArrowDown
+} from '@microsoft/fast-web-utilities';
+import type { Select } from '..';
+import type { ListOption } from '../../list-option';
+import { waitForUpdatesAsync } from '../../testing/async-helpers';
+import { FilterMode } from '../types';
+
+/**
+ * Page object for the `nimble-select` component to provide consistent ways
+ * of querying and interacting with the component during tests.
+ */
+export class SelectPageObject {
+    public constructor(private readonly selectElement: Select) {}
+
+    public async openAndSetFilterText(filterText: string): Promise<void> {
+        if (this.selectElement.filterMode === FilterMode.none) {
+            throw new Error(
+                'Can not set filter text with filterMode set to "none".'
+            );
+        }
+        await this.clickSelect();
+        const filterInput = this.getFilterInput();
+        if (filterInput) {
+            filterInput.value = filterText;
+        }
+
+        await waitForUpdatesAsync();
+        const inputEvent = new InputEvent('input');
+        filterInput?.dispatchEvent(inputEvent);
+        await waitForUpdatesAsync();
+    }
+
+    public async closeDropdown(): Promise<void> {
+        this.selectElement.open = false;
+        await waitForUpdatesAsync();
+    }
+
+    public getFilteredOptions(): ListOption[] {
+        return this.selectElement.filteredOptions as ListOption[];
+    }
+
+    public getSelectedOption(): ListOption | null {
+        return (this.selectElement.selectedOptions[0] as ListOption) ?? null;
+    }
+
+    /**
+     * Either opens or closes the dropdown depending on its current state
+     */
+    public async clickSelect(): Promise<void> {
+        this.selectElement.dispatchEvent(new Event('click'));
+        await waitForUpdatesAsync();
+    }
+
+    public clickSelectedItem(): void {
+        if (!this.selectElement.open) {
+            throw new Error('Select must be open to click selectedItem');
+        }
+
+        this.clickOption(this.selectElement.selectedIndex);
+    }
+
+    public clickOption(index: number): void {
+        if (!this.selectElement.open) {
+            throw new Error('Select must be open to click selectedItem');
+        }
+
+        if (index >= this.selectElement.options.length) {
+            throw new Error(
+                '"index" greater than number of current displayed options'
+            );
+        }
+
+        const option = this.selectElement.options[index]!;
```

Comment:
Does this need to consider the case where the desired option is scrolled off screen? If that's complicated to do but detecting that case is easy, I'd be fine with throwing for now until we have a concrete need for it.

Reply:
Or we could always scroll it into view (though I'm not sure that's even technically necessary). I'm not sure I know of a good reason to prevent a click on an option that is currently scrolled out of view (via this API).

Reply:
I agree scrolling into view would be better, just wasn't sure if there's an easy API for that. But now that I spent 15 seconds researching it, [scrollIntoView()](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) sounds pretty darn promising!